docs/build/guides/dapps/state-archival.mdx

```diff
@@ -0,0 +1,436 @@
+
+```javascript
+async function performContractAction(contractId, method, ...args) {
+  try {
+    const result = await submitOrRestoreAndRetry(contractId, method, ...args);
+    console.log("Transaction successful:", result);
+    return result;
+  } catch (error) {
+    console.error("Error performing contract action:", error);
+    // Handle the error appropriately in your UI
+  }
+}
+```
+
+#### Step 4: Handling contract instance restoration
+
+For restoring an entire contract instance, you might need a separate function:
+
+```javascript
+async function restoreContractInstance(contractId) {
+  try {
+    let hasFreighter = await isConnected();
+    if (!hasFreighter) {
+      return alert("Freighter wallet is required for transactions");
+    }
+
+    const isAllowed = await setAllowed();
+    if (!isAllowed) {
+      return alert("Please allow the transaction in Freighter wallet");
+    }
+
+    const accountId = await getPublicKey();
+    const account = await server.getAccount(accountId);
+    const fee = StellarSdk.BASE_FEE;
+
+    const contract = new StellarSdk.Contract(contractId);
+    const instance = contract.getFootprint();
+    console.log("ins", instance);
+    window.ins = instance;
+
+    // Fetch the WASM entry from the ledger\
+
+    const wasmEntry = await server.getLedgerEntries(instance);
+
+    const restoreTx = new StellarSdk.TransactionBuilder(account, {
+      fee: StellarSdk.BASE_FEE,
+    })
+      .setNetworkPassphrase(StellarSdk.Networks.TESTNET)
+      .setSorobanData(
+        // Set the restoration footprint (remember, it should be in the
+        // read-write part!)
+        new StellarSdk.SorobanDataBuilder()
+          .setReadWrite([
+            instance,
+            ...wasmEntry.entries.map((entry) => entry.key),
+          ])
+          .build(),
+      )
+      .setTimebounds(0, Date.now() + 10000)
+      .addOperation(StellarSdk.Operation.restoreFootprint({}))
+      .build();
+
+    let preparedTx = await server.prepareTransaction(restoreTx);
+    let signedXDR = await signTransaction(preparedTx.toXDR(), {
+      networkPassphrase: networkPassphrase,
+    });
+    let signedTx = StellarSdk.TransactionBuilder.fromXDR(
+      signedXDR,
+      networkPassphrase,
+    );
+
+    return server.sendTransaction(signedTx);
+  } catch (error) {
+    console.error("Error restoring contract instance:", error);
+    throw error;
+  }
+}
+
+// Helper function to get the ledger key for the WASM entry
+function getWasmLedgerKey(entry) {
+  return StellarSdk.xdr.LedgerKey.contractCode(
+    new StellarSdk.xdr.LedgerKeyContractCode({
+      hash: entry.val().instance().wasmHash(),
+    }),
+  );
+}
+```
+
+:::info
+
+This function specifically restores a contract instance and its associated Wasm code. It retrieves the contract's footprint and Wasm entry, creates a restoration transaction, which is then signed using Freighter and submitted to the network.
```

Comment:
It might make this article shorter with less repeats if you cross-referenced this tutorial for this portion: https://developers.stellar.org/docs/data/rpc/api-reference/methods/getLedgerEntries#requesting-a-contracts-wasm-code